### ISSUE TITLE
Consider adding a 'no-wait' argument to commands that may wait on locks

### ISSUE THREAD

**Issue by jgsogo** [conan-io/conan#5500]
TL; DR: Add a no-wait argument, so `conan install .... --no-wait` will exit if it finds a lock in the cache.

----
When the cache is locked Conan prints a nice message and waits _forever_ until the locks are removed. This is usually caused by another process running, but it could also be the **result of a previous command crashing**.

```
> conan install ...

....

fmt/5.3.0@bincrafters/stable is locked by another concurrent conan process, wait...
If not the case, quit, and do 'conan remove --locks'
```

If we were able to pass a `--no-wait` argument, the command would return on this lock condition (maybe with a different exit code) so the caller can inform the user about this circumstance.

This would be very useful for IDE extensions/plugins where the user is not paying attention to the Conan output. If the command doesn't hang we can get the exit code and tell the user what to do next using the IDE windows, error messages,... 

Related to https://github.com/conan-io/conan-vs-extension/issues/144

**Comment by memsharded** [comment]
Yes, but this is also dangerous. The Clion IDE for example installs both debug and release in parallel. If locks are disabled, it will produce race conditions and fail.
Bypassing a mechanism to avoid errors, and leaving it as the default from th IDEs doesn't seem the best approach. I don't know what would be the best UX, but please consider the implications and other possible alternatives.

**Comment by jgsogo** [comment]
No, I'm not talking about bypassing the locks or disabling them. I'm just saying that if I use the `--no-wait` argument the command will finish doing nothing (maybe a message) instead of waiting `forever`.
